lib/user_segments.rb

```diff
@@ -0,0 +1,42 @@
+class UserSegments
+  def self.all_users
+    User.newsletter.active
+  end
+
+  def self.proposal_authors
+    author_ids = Proposal.not_archived.not_retired.pluck(:author_id).uniq
+    all_users.where(id: author_ids)
```

Comment:
This line is the same as line 40 in this file, what makes me think we could delete it and rewrite previous line into just: